File: benchmarks/sum.lua
-- Sum Benchmark - Lua

local function sum_to_n(n)
    local total = 0
    for i = 1, n do
        total = total + i
    end
    return total
end

print("=== Sum Benchmark (Lua) ===")
local N = 1000000

local start = os.clock()
local result = sum_to_n(N)
local elapsed = os.clock() - start

print(string.format("sum(1..%d) = %d", N, result))
print(string.format("Time: %.3fs", elapsed))

File: benchmarks/sum.js
// Sum Benchmark - Node.js

function sumToN(n) {
    let total = 0;
    for (let i = 1; i <= n; i++) {
        total += i;
    }
    return total;
}

console.log("=== Sum Benchmark (Node.js) ===");
const N = 1_000_000;

const start = performance.now();
const result = sumToN(N);
const elapsed = (performance.now() - start) / 1000;

console.log(`sum(1..${N}) = ${result}`);
console.log(`Time: ${elapsed.toFixed(3)}s`);
